src/services/resource.js

```diff
@@ -882,6 +921,89 @@ module.exports = class resourceHelper {
 		return resourceIdsToBeRemoved
 	}
 
+	/**
+	 * Get all resources assigned to the reviewer and already picked up by other reviewer
+	 * @name findResourcesPickedUpByAnotherReviewer
+	 * @param {String} loggedInUserId -  user id of the logged in user.
+	 * @param {Array} finalResourceIds -  list of resources matching to reviewer's level.
+	 * @returns {Array} - Response contain array of resource ids to be removed from the main response
+	 */
+	static async findResourcesSpecificallyAssignedToOtherReviewers(loggedInUserId, finalResourceIds) {
+		// remove all the resouces in sequential review picked up by another reviewer
+		const reviewsFilter = {
+			resource_id: { [Op.in]: finalResourceIds },
+			status: {
+				[Op.in]: [
+					common.REVIEW_STATUS_INPROGRESS,
+					common.REVIEW_STATUS_CHANGES_UPDATED,
+					common.REVIEW_STATUS_REQUESTED_FOR_CHANGES,
+					common.RESOURCE_STATUS_STARTED,
+				],
+			},
+			reviewer_id: { [Op.notIn]: [loggedInUserId] },
+		}
+		const reviewsResponse = await reviewsQueries.findAll(reviewsFilter, ['resource_id'])
+		let resourceIdsToBeRemoved = []
+		if (reviewsResponse) {
+			// push resource ids to resourceIdsToBeRemoved array
+			resourceIdsToBeRemoved = reviewsResponse.map((item) => item.resource_id)
+			return resourceIdsToBeRemoved
+		}
+		return resourceIdsToBeRemoved
+	}
+
+	/**
+	 * Get all resources which reviewer approved and requested for changes
+	 * @name findResourcesCompletedMyReview
+	 * @param {String} loggedInUserId -  user id of the logged in user.
+	 * @param {Array} finalResourceIds -  list of all resources fetched to list.
+	 * @returns {Array} - Response contain array of resource ids to be removed from the main response
+	 */
+	static async findResourcesCompletedMyReview(loggedInUserId, finalResourceIds) {
+		// remove all the resouces from list which reviewer approved and requested for changes
+		const reviewsFilter = {
+			resource_id: { [Op.in]: finalResourceIds },
+			status: {
+				[Op.in]: [common.REVIEW_STATUS_APPROVED, common.REVIEW_STATUS_REQUESTED_FOR_CHANGES],
+			},
+			reviewer_id: loggedInUserId,
+		}
+		const reviewsResponse = await reviewsQueries.findAll(reviewsFilter, ['resource_id'])
+		let resourceIdsToBeRemoved = []
+		if (reviewsResponse) {
```

Comment:
duplicate and length check

Reply:
No duplicates will be here as we are checking reviews table with a specific user id. One user cannot have same resources reviewing multiple times